src/ui/page/share/dataset/shareDataset.tsx

```diff
@@ -5,15 +5,48 @@ import { DataSelection } from './steps/dataSelection/dataSelection'
 import { MetadataFilling } from './steps/metadataFilling/metadataFilling'
 import { Stepper } from '@/ui/component/stepper/stepper'
 import type { StepElement } from '@/ui/component/stepper/stepper'
+import { useAppStore } from '@/ui/store'
+import type { DatePickerValue, FormItem } from '@/ui/store/slice/shareData/shareData.slice'
 import '../i18n/index'
 import './shareDataset.scss'
 
+const isFormItemFilled = (item: FormItem): boolean => {
+  switch (item.type) {
+    case 'text': {
+      return !!(item.value as unknown as string)
+    }
+    case 'select': {
+      return !!(item.value as unknown as string[]).length
+    }
+    case 'numeric': {
+      const numericField = item.value as unknown as number
+      return numericField === 0 || !!numericField
+    }
+    case 'date': {
+      const datePickerField = item.value as unknown as DatePickerValue
+      return !!datePickerField.from || !!datePickerField.to
+    }
+  }
+}
```

Comment:
😵

 I'm a little too late, but the problem here lies in the following type definitions:

```ts
type FormItemType = 'text' | 'numeric' | 'date' | 'select'
type FormItemValue = string | string[] | number | DatePickerValue

type FormItem = Readonly<{
  id: FormItemId
  type: FormItemType
  label: string
  value: FormItemValue
  required?: boolean
}>
```

This definitely illustrates the problem with how we're defining the `FormItem` type, which does not embrace an [Algebraic Data Type](https://en.wikipedia.org/wiki/Algebraic_data_type) approach. Firstly, we are allowing representations that should not be possible, which can lead to invalid states (rule:  [Making illegal states unrepresentable](https://fsharpforfunandprofit.com/posts/designing-with-types-making-illegal-states-unrepresentable/)). Secondly, we're not leveraging discriminated unions, which would allow us to safely manipulate the values in a type-safe manner.

Reply:
Good point. We could update FormItem time as follow:
```ts
type FormItem = {
  type: 'text';
  value: string;
} | {
  type: 'select';
  value: string[];
} | {
  type: 'numeric';
  value: number;
} | {
  type: 'date';
  value: DatePickerValue;
} | ...
```

Thus simplify the `isFormItemFilled ` function:
```ts
const isFormItemFilled = (item: FormItem): boolean => {
  switch (item.type) {
    case 'text': {
      return !!item.value
    }
    case 'select': {
      return item.value.length > 0
    }
    case 'numeric': {
      return item.value === 0 || !!item.value
    }
    case 'date': {
      return !!item.value.from || !!item.value.to
    }
  }
}
```

Reply:
Oh my! Exactly what should be done @0xLodz :relieved:  What I am reading is a delight to my eyes and fills me with an ineffable joy.

Reply:
Indeed it's much more elegant and safer, well seen @ccamel  and well coded @0xLodz !